src/main/scala/io/iohk/ethereum/mining/BlockGenerator.scala

```diff
@@ -76,6 +64,32 @@ class BlockGenerator(blockchainStorages: BlockchainStorages, blockchainConfig: B
     result
   }
 
+  private def prepareTransactions(transactions: Seq[SignedTransaction], blockGasLimit: BigInt) = {
+    val sortedTransactions = transactions.groupBy(_.senderAddress).values.toList.flatMap { txsFromSender =>
+      val ordered = txsFromSender
+        .sortBy(-_.tx.gasPrice)
+        .sortBy(_.tx.nonce)
+        .scanLeft(None: Option[BigInt], None: Option[SignedTransaction]) { case ((prevNonce, _), stx) =>
```

Comment:
yes, I think it will be easier to read but it's not mandatory your change. If you consider it's better, go ahead.